### PR TITLE
docs: Add docs for state backend migration

### LINKED ISSUE #6996: docs: Add docs for how to copy state to a new backend
Using existing interfaces and features, can we provide a guide for how to copy commands from one backend to a new one?

### PR DESCRIPTION
**Pull request by cjohnhanson**
Closes #6996

### CODE REVIEW

**Review by afolson — APPROVED** [review]

File: docs/src/_concepts/state_backends.md [hunk 1]
```diff
@@ -5,56 +5,85 @@ layout: doc
...
-To cut down on extraneous dependencies, Meltano imports cloud storage client libraries lazily and these libraries are not included in the base `meltano` package.
+No extra work is needed to use the default system database or local filesystem as a state backend as they are already part of any base Meltano distribution.
 
-Only the default system database state backend _or_ the local filesystem state backend can be used as part of the base `meltano` distribution.
+To use a cloud storage backend, install Meltano using one of the following [extras](https://peps.python.org/pep-0508/#extras):
 
-To use a cloud storage backend, install Meltano using one of the following (extras)[https://peps.python.org/pep-0508/#extras]:
 - `meltano[s3]` to use the AWS S3 state backend.
 - `meltano[azure]` to use the Azure Blob Storage state backend.
 - `meltano[gcs]` to use the Google Cloud Storage state backend.
```
Comment (tayloramurphy) [inline docs/src/_concepts/state_backends.md:30]: Do we have guidance on what to do if a user has already installed Meltano? Is it just a reinstall? Should they do something else?

File: docs/src/_concepts/state_backends.md [hunk 2]
```diff
@@ -5,56 +5,85 @@ layout: doc
...
+If these environment variables are not set, it will use the credentials stored in the environment's [AWS credentials file](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html).
 
 If AWS credentials are not found via any of the methods described above, Meltano will not be able to authenticate to S3 and state operations will fail.
 
-
 ### Google Cloud Storage
+
 To store state remotely in Google Cloud Storage, set the `state_backend.uri` setting to `gs://<your bucket name>/<prefix for state JSON blobs>`.
 
 To authenticate to GCS, you must provide a path to a (service account credentials file)[https://cloud.google.com/iam/docs/creating-managing-service-account-keys]
```
Comment (tayloramurphy) [inline docs/src/_concepts/state_backends.md:89]:
```suggestion
To authenticate to GCS, you must provide a path to a [service account credentials file](https://cloud.google.com/iam/docs/creating-managing-service-account-keys).
```

**Review by tayloramurphy — CHANGES_REQUESTED** [review]
@cjohnhanson I did some refactoring on this to smooth out the flow - but the vast majority of the content was fantastic. I never got a chance to review the original PR for the docs so I did that here as well. Just one question for you and then I think we can merge this!

File: docs/src/_concepts/state_backends.md (hunk 1 above)
Comment (cjohnhanson) [reply docs/src/_concepts/state_backends.md:30]: Good callout. The user would just need to install the relevant extra (e.g. `pip install "meltano[s3]"` and pip will only install the missing dependencies.

I also added a callout that the user will need to `meltano upgrade` to 2.10+ in order to use state backends.

**Review by tayloramurphy — APPROVED** [review]
Looks great! Love the addition. :shipit:

**Pull request merged**